studio/components/interfaces/Realtime/Inspector/useRealtimeMessages.ts

```diff
@@ -0,0 +1,214 @@
+    // Need to add 'extension' key in the 'payload'
+    newChannel.on('system' as any, {} as any, (payload: any) => {
+      // if (payload.extension === 'postgres_changes' && payload.status === 'ok') {
+      //   pushMessageTo('#conn_info', 'postgres_subscribed', {})
+      // }
+      pushMessage('SYSTEM', payload)
+    })
+
+    if (enableBroadcast) {
+      // Listen for all (`*`) `broadcast` messages
+      // The message name can by anything
+      // Match on specific message names to filter for only those types of messages and do something with them
+      newChannel.on('broadcast', { event: '*' }, (payload) => pushMessage('BROADCAST', payload))
+    }
+
+    // Listen for all (`*`) `presence` messages
+    if (enablePresence) {
+      newChannel.on('presence' as any, { event: '*' }, (payload) => {
+        pushMessage('PRESENCE', payload)
+      })
+    }
+
+    if (enableDbChanges) {
+      let postgres_changes_opts: any = {
+        event: '*',
+        schema: schema,
+        table: table,
+        filter: undefined,
+      }
+      if (filter !== '') {
+        postgres_changes_opts.filter = filter
+      }
+      newChannel.on('postgres_changes' as any, postgres_changes_opts, (payload: any) => {
+        let ts = performance.now() + performance.timeOrigin
+        let payload_ts = Date.parse(payload.commit_timestamp)
+        let latency = ts - payload_ts
+        pushMessage('POSTGRES', { ...payload, latency })
+      })
+    }
+
+    // Finally, subscribe to the Channel we just setup
+    newChannel.subscribe(async (status, error) => {
+      if (status === 'SUBSCRIBED') {
+        // Let LiveView know we connected so we can update the button text
+        // pushMessageTo('#conn_info', 'broadcast_subscribed', { host: host })
+
+        if (enablePresence) {
+          const name = 'user_name_' + Math.floor(Math.random() * 100)
+          newChannel.send({
+            type: 'presence',
+            event: 'TRACK',
+            payload: { name: name, t: performance.now() },
+          })
+        }
+      } else if (status === 'CLOSED') {
+        // console.log(`Realtime Channel status: ${status}`)
+      } else {
+        // console.error(`Realtime Channel error status: ${status}`)
+        // console.error(`Realtime Channel error: ${error}`)
+      }
+    })
+
+    setChannel(newChannel)
+    return () => {
+      newChannel.unsubscribe()
+      setChannel(undefined)
+    }
+  }, [
+    client,
+    channelName,
+    enableBroadcast,
+    enableDbChanges,
+    enablePresence,
+    filter,
+    host,
+    schema,
+    table,
+  ])
+
+  const sendMessage = useCallback(
+    (message: string, payload: any) => {
+      if (channel) {
+        channel.send({
+          type: 'broadcast',
+          event: message,
+          payload,
+        })
+        toast.success('Successfully broadcasted message')
+      } else {
+        toast.error('Failed to broadcast message: channel as not been set')
```

Comment:
```suggestion
        toast.error('Failed to broadcast message: channel has not been set')
```
I'm also seeing this error when trying to send a test message with a channel set:
<img width="587" alt="Screenshot 2023-11-06 at 16 27 43" src="https://github.com/supabase/supabase/assets/10985857/db72f15c-3b98-493e-be04-ff416efd0ad8">